libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -979,17 +979,25 @@ AP_GPS_UBLOX::_parse_gps(void)
                         gnssCount++;
                     }
                 }
-
                 for(int i = 0; i < _buffer.gnss.numConfigBlocks; i++) {
-                    // Reserve an equal portion of channels for all enabled systems
+                    // Reserve an equal portion of channels for all enabled systems that supports it
                     if(gps._gnss_mode[state.instance] & (1 << _buffer.gnss.configBlock[i].gnssId)) {
-                        if(GNSS_SBAS !=_buffer.gnss.configBlock[i].gnssId) {
+
+                        if(GNSS_SBAS !=_buffer.gnss.configBlock[i].gnssId && (supports_F9_config() || is_ublox_M9() || GNSS_GALILEO !=_buffer.gnss.configBlock[i].gnssId)) {
                             _buffer.gnss.configBlock[i].resTrkCh = (_buffer.gnss.numTrkChHw - 3) / (gnssCount * 2);
                             _buffer.gnss.configBlock[i].maxTrkCh = _buffer.gnss.numTrkChHw;
-                        } else {
-                            _buffer.gnss.configBlock[i].resTrkCh = 1;
-                            _buffer.gnss.configBlock[i].maxTrkCh = 3;
                         }
+                        else {
+                            if(GNSS_SBAS ==_buffer.gnss.configBlock[i].gnssId) {
+                                _buffer.gnss.configBlock[i].resTrkCh = 1;
+                                _buffer.gnss.configBlock[i].maxTrkCh = 3;
+                            }
+                            if( (!supports_F9_config() && !is_ublox_M9()) && GNSS_GALILEO ==_buffer.gnss.configBlock[i].gnssId) {
```

Comment:
Why do you have to check the version again here? It should be the only way we get to this path....

Reply:
Fixed